superchain/superchain_test.go

```diff
@@ -281,3 +281,18 @@ func TestContractBytecodes(t *testing.T) {
 		}
 	}
 }
+
+// TestCanyonTimestampOnBlockBoundary asserts that Canyon will activate on a block's timestamp.
+// This is critical because the create2Deployer only activates on a block's timestamp.
+func TestCanyonTimestampOnBlockBoundary(t *testing.T) {
+	for id, config := range OPChains {
+		if config.CanyonTime == nil {
+			continue // skip if canyon is not set
+		}
+		canyonOffset := *config.CanyonTime - config.Genesis.L2Time
+		if canyonOffset%2 != 0 {
```

Comment:
nit: can we add a `const BlockTime` to the `superchain` package and use it instead of magic `2`s?

Reply:
would be nice. We also hardcode it in op-node which is why I went with the hardcode here (b/c there was nowhere to import it from)

Reply:
We should update the config in `op-node` so that if blocktime is `nil` it defaults to a constant value (`2s`), then this constant can be imported everywhere. This reduces required config and can be done for the other config values that have superchain required defaults

Reply:
upstream const that can be imported everywhere would definitely be even better! but I thought keeping this change localized to the `superchain-registry` repo and add it locally would still be better than continue hardcoding consts.